sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileBasedSinkTest.java

```diff
@@ -417,6 +419,89 @@ public void testGenerateOutputFilenamesWithoutExtension() {
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testBuildTemporaryDirectoryName() {
+    // Tests for local files without the scheme.
+    assertThat(
+        new FileBasedWriteOperation.TemporaryDirectoryBuilder()
+            .apply("/home/output"),
+        Matchers.startsWith("/home/temp-beam-output"));
+    assertThat(
+        new FileBasedWriteOperation.TemporaryDirectoryBuilder()
+            .apply("/home/output/"),
+        Matchers.startsWith("/home/output/temp-beam--"));
```

Comment:
It might be good to special-case this case where there is no base filename, to avoid producing the confusing `--` - it looks like a bug even though it isn't.